File: 프로그래머스/lv1/12901. 2016년/2016년.py
'''
2016년 a월 b일은 무슨 요일?
29일 - 2
30일 - 4, 6, 9, 11
31일 - 1, 3, 5, 7, 8, 10, 12

'''
def solution(a, b):
    answer = ''
    month_dic = {1:31,2:29,3:31,4:30,5:31,6:30,7:31,8:31,9:30,10:31,11:30,12:31}
    day_sum = 0
    for i in range(1,a):
        day_sum += month_dic.get(i)
    day_sum += b
    
    day = day_sum % 7
    
    if(day == 1):
        answer += "FRI"
    elif(day == 2):
        answer += "SAT"
    elif(day == 3):
        answer += "SUN"
    elif(day == 4):
        answer += "MON"
    elif(day == 5):
        answer += "TUE"
    elif(day == 6):
        answer += "WED"
    elif(day == 0):
        answer += "THU"
    
    
    return answer

File: 프로그래머스/1/12901. 2016년/2016년.java
import java.util.*;
class Solution {
    public String solution(int a, int b) {
        String answer = "";
        HashMap<Integer, Integer> hs = new HashMap<>();
        hs.put(1, 31);
        hs.put(2, 29);
        hs.put(3, 31);
        hs.put(4, 30);
        hs.put(5, 31);
        hs.put(6,30);
        hs.put(7,31);
        hs.put(8,31);
        hs.put(9,30);
        hs.put(10,31);
        hs.put(11,30);
        hs.put(12,31);
        int day_sum = 0;

        for(int i = 1; i < a; i++){
            day_sum += hs.get(i);
        }
        day_sum += b;

        int day = day_sum % 7;

        switch(day){
            case 0:
                answer += "THU";
                break;
            case 1:
                answer += "FRI";
                break;
            case 2:
                answer += "SAT";
                break;
            case 3:
                answer += "SUN";
                break;
            case 4:
                answer += "MON";
                break;
            case 5:
                answer += "TUE";
                break;
            case 6:
                answer += "WED";
                break;
        }
        return answer;
        }
    }
